orchagent/copporch.cpp

```diff
@@ -0,0 +1,488 @@
+    {"lacp", SAI_HOSTIF_TRAP_ID_LACP},
+    {"eapol", SAI_HOSTIF_TRAP_ID_EAPOL},
+    {"lldp", SAI_HOSTIF_TRAP_ID_LLDP},
+    {"pvrst", SAI_HOSTIF_TRAP_ID_PVRST},
+    {"igmp_query", SAI_HOSTIF_TRAP_ID_IGMP_TYPE_QUERY},
+    {"igmp_leave", SAI_HOSTIF_TRAP_ID_IGMP_TYPE_LEAVE},
+    {"igmp_v1_report", SAI_HOSTIF_TRAP_ID_IGMP_TYPE_V1_REPORT},
+    {"igmp_v2_report", SAI_HOSTIF_TRAP_ID_IGMP_TYPE_V2_REPORT},
+    {"igmp_v3_report", SAI_HOSTIF_TRAP_ID_IGMP_TYPE_V3_REPORT},
+    {"sample_packet", SAI_HOSTIF_TRAP_ID_SAMPLEPACKET},
+    {"switch_cust_range", SAI_HOSTIF_TRAP_ID_SWITCH_CUSTOM_RANGE_BASE},
+    {"arp_req", SAI_HOSTIF_TRAP_ID_ARP_REQUEST},
+    {"arp_resp", SAI_HOSTIF_TRAP_ID_ARP_RESPONSE},
+    {"dhcp", SAI_HOSTIF_TRAP_ID_DHCP},
+    {"ospf", SAI_HOSTIF_TRAP_ID_OSPF},
+    {"pim", SAI_HOSTIF_TRAP_ID_PIM},
+    {"vrrp", SAI_HOSTIF_TRAP_ID_VRRP},
+    {"bgp", SAI_HOSTIF_TRAP_ID_BGP},
+    {"dhcpv6", SAI_HOSTIF_TRAP_ID_DHCPV6},
+    {"ospvfv6", SAI_HOSTIF_TRAP_ID_OSPFV6},
+    {"vrrpv6", SAI_HOSTIF_TRAP_ID_VRRPV6},
+    {"bgpv6", SAI_HOSTIF_TRAP_ID_BGPV6},
+    {"neigh_discovery", SAI_HOSTIF_TRAP_ID_IPV6_NEIGHBOR_DISCOVERY},
+    {"mld_v1_v2", SAI_HOSTIF_TRAP_ID_IPV6_MLD_V1_V2},
+    {"mld_v1_report", SAI_HOSTIF_TRAP_ID_IPV6_MLD_V1_REPORT},
+    {"mld_v2_done", SAI_HOSTIF_TRAP_ID_IPV6_MLD_V1_DONE},
+    {"mld_v2_report", SAI_HOSTIF_TRAP_ID_MLD_V2_REPORT},
+    {"ip2me", SAI_HOSTIF_TRAP_ID_IP2ME},
+    {"ssh", SAI_HOSTIF_TRAP_ID_SSH},
+    {"snmp", SAI_HOSTIF_TRAP_ID_SNMP},
+    {"router_custom_range", SAI_HOSTIF_TRAP_ID_ROUTER_CUSTOM_RANGE_BASE},
+    {"l3_mtu_error", SAI_HOSTIF_TRAP_ID_L3_MTU_ERROR},
+    {"ttl", SAI_HOSTIF_TRAP_ID_TTL_ERROR}
+};
+
+std::map<string, sai_packet_action_t> packet_action_map = {
+    {"drop", SAI_PACKET_ACTION_DROP},
+    {"forward", SAI_PACKET_ACTION_FORWARD},
+    {"copy", SAI_PACKET_ACTION_COPY},
+    {"copy_cancel", SAI_PACKET_ACTION_COPY_CANCEL},
+    {"trap", SAI_PACKET_ACTION_TRAP},
+    {"log", SAI_PACKET_ACTION_LOG},
+    {"deny", SAI_PACKET_ACTION_DENY},
+    {"transit", SAI_PACKET_ACTION_TRANSIT}
+};
+
+CoppOrch::CoppOrch(DBConnector *db, string tableName) :
+    Orch(db, tableName)
+{
+    SWSS_LOG_ENTER();
+};
+
+bool CoppOrch::getTrapID(string &trap_id_str, sai_hostif_trap_id_t &trap_id) const
+{
+    SWSS_LOG_ENTER();
+    auto it = trap_id_map.find(trap_id_str);
+    if (it == trap_id_map.end())
+    {
+        SWSS_LOG_ERROR("Invalid trap_id specified %s\n", trap_id_str.c_str());
+        return false;
+    }
+    trap_id = trap_id_map[trap_id_str];
+    SWSS_LOG_DEBUG("Output value:%d", trap_id);
+    return true;
+}
+
+bool CoppOrch::getTrapIdList(vector<string> &trap_id_name_list, vector<sai_hostif_trap_id_t> &trap_id_list) const
+{
+    SWSS_LOG_ENTER();
+    for(auto trap_id_str : trap_id_name_list)
+    {
+        sai_hostif_trap_id_t trap_id;
+        if (!getTrapID(trap_id_str, trap_id))
+        {
+            SWSS_LOG_ERROR("Invalid trap_id value specified %s\n", trap_id_str.c_str());
```

Comment:
this error message will appear twice both in getTrapID function and here. Is it better to combine these two functions and just use try/catch logic to make it simpler?